src/services/api/data.js

```diff
@@ -0,0 +1,43 @@
+import defaultClient from '@/services/api/defaultClient.js';
+import endpoint from '@/services/api/endpoint.js';
+
+export async function getAllMoviesData() {
+  const moviesResponse = await defaultClient.get(endpoint.MOVIES);
+  return moviesResponse.data;
+}
+
+export async function getMovieData(movieId) {
+  const movieResponse = await defaultClient.get(`${endpoint.MOVIES}/${movieId}`);
+  return movieResponse.data;
+}
+
+export async function getMovieGenresData() {
+  const genreResponse = await defaultClient.get(`${endpoint.GENRES}`);
+  return genreResponse.data;
+}
+
+export async function getSeancesData(movieId = null, date = null) {
+  const hasMovieIdQuery = movieId !== null;
+  const hasDateQuery = date !== null;
+  const hasAllPossibleQueries = hasMovieIdQuery && hasDateQuery;
+
+  let exactEndpoint;
+
+  if (hasAllPossibleQueries) {
+    exactEndpoint = `${endpoint.SEANCES}?movie_id=${movieId}&date=${date}`;
+  } else if (hasMovieIdQuery) {
+    exactEndpoint = `${endpoint.SEANCES}?movie_id=${movieId}`;
+  } else if (hasDateQuery) {
+    exactEndpoint = `${endpoint.SEANCES}?date=${date}`;
+  } else {
+    exactEndpoint = `${endpoint.SEANCES}`;
+  }
+
+  const screeningsResponse = await defaultClient.get(exactEndpoint);
```

Comment:
Just leave it to axios to handle this and **leave the default values as undefined** 😎:
```suggestion
  const screeningsResponse = await defaultClient.get(exactEndpoint, {
    params: {
      movie_id: movieId,
      date: date
    }
  });
```
This way any undefined value will not be added to query and there is no need to have all this logic here.

Reply:
Seems good that I've changed a little bit folder structure in `/services`, because all of that code have passed previous PRs and otherwise I would miss such an important tip, thank you!